scripts/tools/generate_esp32_chip_factory_bin.py

```diff
@@ -277,40 +277,48 @@ def gen_spake2p_params(passcode):
 
 
 def populate_factory_data(args, spake2p_params):
-    FACTORY_DATA['discriminator']['value'] = args.discriminator
-    FACTORY_DATA['iteration-count']['value'] = spake2p_params['Iteration Count']
-    FACTORY_DATA['salt']['value'] = spake2p_params['Salt']
-    FACTORY_DATA['verifier']['value'] = spake2p_params['Verifier']
-    FACTORY_DATA['dac-cert']['value'] = os.path.abspath(args.dac_cert)
-    FACTORY_DATA['pai-cert']['value'] = os.path.abspath(args.pai_cert)
-    FACTORY_DATA['cert-dclrn']['value'] = os.path.abspath(args.cd)
-    FACTORY_DATA['dac-key']['value'] = os.path.abspath('dac_raw_privkey.bin')
-    FACTORY_DATA['dac-pub-key']['value'] = os.path.abspath('dac_raw_pubkey.bin')
-
-    if args.serial_num is not None:
+    if args.discriminator:
```

Comment:
can discriminator be 0? I think for numbers you may need to check for None since both zero and non-zero should be acceptable.

Reply:
Yes, discriminator can be zero, so added back the check against `None` for this as well as other integer values.